### DISCUSSION TITLE
Bug in Motor Block (does not set

### DISCUSSION THREAD

**Discussion by mperino** [BIPES/BIPES#234] [General]
Maker Pi RP2040 
Using the following block code:
![image](https://github.com/user-attachments/assets/db1236b3-9fd0-4dda-9709-61c616e622a8)

Which Yields the following Python:

=== from machine import Pin
=== from machine import PWM
=== import time
=== 
=== 
=== #Code automatically generated by BIPES (http://www.bipes.net.br)
=== #Author: 'mperino'
=== #IOT ID: 0
=== #Description: 'Sound and Motor Demo'
=== 
=== FULL_POWER_LEVEL = 65024
=== right_forward = PWM(Pin(8))
=== right_reverse = PWM(Pin(9))
=== left_forward = PWM(Pin(10))
=== left_reverse = PWM(Pin(11))
=== 
=== def forward():
===     right_reverse.duty_u16(0)
===     left_reverse.duty_u16(0)
===     right_forward.duty_u16(FULL_POWER_LEVEL)
===     left_forward.duty_u16(FULL_POWER_LEVEL)
=== 
=== def forwardSlow():
===     right_reverse.duty_u16(0)
===     left_reverse.duty_u16(0)
===     right_forward.duty_u16(FULL_POWER_LEVEL // 2)
===     left_forward.duty_u16(FULL_POWER_LEVEL // 2)
=== 
=== def reverse():
===     right_forward.duty_u16(0)
===     left_forward.duty_u16(0)
===     right_reverse.duty_u16(FULL_POWER_LEVEL)
===     left_reverse.duty_u16(FULL_POWER_LEVEL)
=== 
=== def reverseSlow():
===     right_forward.duty_u16(0)
===     left_forward.duty_u16(0)
===     right_reverse.duty_u16(FULL_POWER_LEVEL // 2)
===     left_reverse.duty_u16(FULL_POWER_LEVEL // 2)
=== 
=== def left():
===     left_forward.duty_u16(0)
===     right_reverse.duty_u16(0)
===     left_reverse.duty_u16(FULL_POWER_LEVEL // 2)
===     right_forward.duty_u16(FULL_POWER_LEVEL // 2)
=== 
=== def right():
===     right_forward.duty_u16(0)
===     left_reverse.duty_u16(0)
===     right_reverse.duty_u16(FULL_POWER_LEVEL // 2)
===     left_forward.duty_u16(FULL_POWER_LEVEL // 2)
=== 
=== def stop():
===     right_forward.duty_u16(0)
===     left_forward.duty_u16(0)
===     right_reverse.duty_u16(0)
===     left_reverse.duty_u16(0)
=== 
=== stop()
=== forward()
=== time.sleep(1)
=== stop()
=== time.sleep(1)
=== reverseSlow()
=== time.sleep(1)
=== stop()
=== 
=== 
=== 

Never sets a frequency for the PWM.
`right_forward.freq(50)`
is needed (and same for the other side and direction).
I don't see any block to set it either.

**Comment by mperino** [comment]
Im using:
https://bipes.net.br/pico/ui/

**Comment by jetannenbaum** [comment]
The blocks work for me.  I am using MicroPython version 1.20.0  A few revs behind the latest release on my test setup.  What version of MicroPython are you using?

**Comment by jetannenbaum** [comment]
This might explain the issue.  Version 1.21.0 has the following in the release notes: "rp2: machine.PWM enhancements"  No explanation of what the enhancements might be.  I'll reflash my MakerPi board and test with the latest version.

**Comment by jetannenbaum** [comment]
Went a little further in the release notes for version 1.21.0.  Found this: machine_pwm: start PWM only if freq and duty are set.  Looks like I need to set the freq (as you suggested).  I'll update the code and release as soon as I can get my power to stop cycling.  Stay tuned for an update.

**Comment by jetannenbaum** [comment]
Confirmed the bug with MicroPython version 1.23.0.  I update the bipes repo and code on the server.  Confirmed the new code works in version 1.20.1 and 1.23.0.  Please make sure to flush the cache before trying the now code.  Thanks for your support

**Comment by mperino** [comment]
Wow you are speedy!  Works Perfectly!
